### PR TITLE
Enhancement/6243 loyal visitors

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #6243 

## Relevant technical choices

* Updated the MetricTileNumeric component to allow percentage styles.
* Updated the New Visitors widget to pull report data only when it's needed.
* Updated VRT images for existing Key Metrics related components to share the same prefix.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @eugene-manuilov – this looks good to me, just a few points to consider.

File: assets/js/util/i18n.js [hunk 1]
```diff
@@ -262,7 +296,7 @@ export const readableLargeNumber = ( number ) => {
  *                                                      or a currency code.
  * @return {string} The formatted number.
  */
-export const numFmt = ( number, options = {} ) => {
+export function numFmt( number, options = {} ) {
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/util/i18n.js:299]: There is no benefits in using arrow functions for top level functions. IMO it just makes it harder to read. Created a ticket to add an eslint rule: #7146

File: assets/js/components/KeyMetrics/MetricTileNumeric.js [hunk 2]
```diff
@@ -24,18 +24,22 @@ import PropTypes from 'prop-types';
...
+		metricValue,
+		metricValueFormat,
+		subText,
+		previousValue,
+		currentValue,
+	} = props;
+
 	if ( loading ) {
 		return (
 			<Widget noPadding>
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/components/KeyMetrics/MetricTileNumeric.js:45]: Updated.

File: assets/js/util/i18n.js [hunk 3]
```diff
@@ -245,6 +245,40 @@ export const readableLargeNumber = ( number ) => {
...
 
+/**
+ * Parses formatting options and returns an object with options for selected formatting.
+ *
+ * @since n.e.x.t
+ *
+ * @param {(Intl.NumberFormatOptions|string)} options Formatting options or unit shorthand. Possible shorthand values are '%', 's', or a currency code.
+ * @return {Object} Formatting options.
+ */
+export function numFmtOptions( options ) {
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/util/i18n.js]: Renamed.

File: assets/js/modules/analytics-4/components/widgets/NewVisitorsWidget.js [hunk 4]
```diff
@@ -44,34 +44,35 @@ export default function NewVisitorsWidget( { Widget, WidgetNull } ) {
...
 
-	const dates = useSelect( ( select ) =>
-		select( CORE_USER ).getDateRangeDates( {
-			offsetDays: DATE_RANGE_OFFSET,
-			compare: true,
-		} )
-	);
+	// One combined select hook is used to prevent unnecessary selects
+	// if the key metrics widget is hidden.
+	const [ report, loading ] =
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics-4/components/widgets/NewVisitorsWidget.js]: Good idea. Added a new HOC.

**Review by aaemnnosttv — APPROVED** [review]
Great, thanks @eugene-manuilov !

**Pull request merged**